indexer-app/src/cmr/indexer/data/concept_parser.clj

```diff
@@ -57,11 +58,26 @@
   [context concept]
   (edn/read-string (:metadata concept)))
 
-(doseq [concept-type (concepts/get-generic-concept-types-array)]
+(doseq [concept-type concepts/get-generic-non-draft-concept-types-array]
   (defmethod parse-concept concept-type
     [context concept]
     (json/parse-string (:metadata concept) true)))
 
+(doseq [concept-type concepts/get-draft-concept-types-array]
+  (defmethod parse-concept concept-type
+    [context concept]
+    ;; If the draft record is a json record, then parse it,
+    ;; otherwise figure out what the draft record concept type is
+    ;; and let the parsing code parse it to umm-c.
+    (if (string/includes? (:format concept) "json")
+      (json/parse-string (:metadata concept) true)
+      (let [concept-type (:concept-type concept)]
+        (if (concepts/is-draft-concept? concept-type)
```

Comment:
Should we pull this `(if (concepts/is-draft-concept? concept-type)` to the top of the function to check if the concept is a draft type before this check: `(if (string/includes? (:format concept) "json")`